### ISSUE TITLE
Video starts playing after the whole file is downloaded

### ISSUE THREAD

**Issue by martinflorek** [neekeetab/CachingPlayerItem#12]
I want to start video playback as soon as possible instead of waiting for the whole file to be downloaded first.

I set the `AVPLayer`'s `automaticallyWaitsToMinimizeStalling` to `false` and `AVPlayerItem`'s `preferredForwardBufferDuration` to different values ranging from 0.1 to 2.0 but the AVPLayer always waits for the full download. I also tried `AVPLayer`'s `playImmediately(atRate: 1.0)` method instead of the plain `play()`.

The KVO observed `status` in the `CachingPlayerItem` is changed to "ready" only after the data is fully downloaded. 

How can I start video playback immediately after some data is downloaded instead of waiting for the full download?

**Comment by martinflorek** [comment]
@neekeetab thanks for the quick reply. I just tried it with mp3 audio and the playback starts almost instantly and starts before the full download. The status is "ready" after some kilobytes are downloaded and the audio starts playing. So this looks like a a video-only issue.

I tried this on a real device and both the mp3 and mp4 files are hosted on the same server with same permissions. Both HTTP and HTTPS connections behave the same.

I also tried the Big Buck Bunny video and its playback starts almost instantly and the download continues in the background. So it looks like encoding parameters issue. I will play with those now.
